client/lib/gsuite/new-users.ts

```diff
@@ -351,15 +354,22 @@ const getItemsForCart = (
 	productSlug: string,
 	users: GSuiteNewUser[]
 ): MinimalRequestCartProduct[] => {
-	const usersGroupedByDomain: { [ domain: string ]: GSuiteProductUser[] } = mapValues(
-		groupBy( users, 'domain.value' ),
-		( groupedUsers ) => groupedUsers.map( transformUserForCart )
+	const usersGroupedByDomain = users.reduce(
+		( usersByDomain: Record< string, GSuiteProductUser[] >, user: GSuiteNewUser ) => {
```

Comment:
I find having `usersGroupedByDomain` and `usersByDomain` side-by-side confusing (the first question that comes to my mind when seeing this is what's the difference between the two?). Something that I often do with `reduce()` functions is to use a generic name like `result` for the first parameter, and I think this could help with reducing this confusion here (and with `workingCount` above):

```suggestion
		( result: Record< string, GSuiteProductUser[] >, user: GSuiteNewUser ) => {
```

Reply:
Done! 16db851